warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileRecordReader.java

```diff
@@ -0,0 +1,109 @@
+package datawave.ingest.mapreduce.job;
+
+import static org.apache.accumulo.core.conf.Property.TABLE_CRYPTO_PREFIX;
+
+import java.io.EOFException;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Collections;
+
+import org.apache.accumulo.core.crypto.CryptoFactoryLoader;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.file.FileSKVIterator;
+import org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile;
+import org.apache.accumulo.core.file.rfile.RFile;
+import org.apache.accumulo.core.spi.crypto.CryptoEnvironment;
+import org.apache.accumulo.core.spi.crypto.CryptoService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+public class SplittableRFileRecordReader extends RFileRecordReader {
+    /**
+     * Initializes the fileIterator from the split. If split is an RFileSplit the record reader will create an iterator that only covers the content for the
+     * split. If instead the split is a FileSplit it will be handled by {@link RFileRecordReader}.
+     *
+     * @param split
+     * @param context
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Override
+    public void initialize(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+        if (split instanceof RFileSplit) {
+            fileIterator = getIterator(context.getConfiguration(), (RFileSplit) split);
+        } else {
+            super.initialize(split, context);
+        }
+    }
+
+    public static RFile.Reader getRFileReader(Configuration config, Path rfile) throws IOException {
+        FileSystem fs = rfile.getFileSystem(config);
+
+        if (!fs.exists(rfile)) {
+            throw new FileNotFoundException(rfile + " does not exist");
+        }
+
+        CryptoService cs = CryptoFactoryLoader.getServiceForClient(CryptoEnvironment.Scope.TABLE, config.getPropsWithPrefix(TABLE_CRYPTO_PREFIX.name()));
+        CachableBlockFile.CachableBuilder cb = new CachableBlockFile.CachableBuilder().fsPath(fs, rfile).conf(config).cryptoService(cs);
```

Comment:
This is some 'magic code' that's reused in the unit test, perhaps a good candidate for a static utility method?